### ISSUE TITLE
Rest arguments in a generic interface lead to Internal error

### ISSUE THREAD

**Issue by futpib** [facebook/flow#5437]
Both of the following examples fail with the following message

```
Internal error: Failure(Did not expect BoundT)
```

1. [try flow link](https://flow.org/try/#0FASwdgLgpgTgZgQwMZQAQElK0SgPAQQD5UBvYVVAWyggAsB7AEwAoA6d-ASgC5UA3eiEYBuYAF8gA)

```js
interface Interface<A> {
  method(...A): void;
}
```

2. [try flow link](https://flow.org/try/#0FASwdgLgpgTgZgQwMZQAQElK0SgPAQQC5V8YYEBPXBMCgPjtQG9hVUBbKCACwHsATABQA6UfgCUxAG68Q-ANzAAvkA)

```js
interface Interface<A: Array<any>> {
  method(...A): void;
}
```

If you remove `...` everything works.

**Comment by julienw** [comment]
Still present today

**Comment by julienw** [comment]
My code:
```javascript
interface JestMockFn<TArguments> {
  (...args: TArguments): void;
}
```

The error:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ test.js:0:1

Internal error: uncaught exception: Failure("Did not expect BoundT")
Raised at file "map.ml", line 130, characters
10-25
Called from file "hack/utils/collections/myMap.ml", line 15, characters 13-23




Found 1 error
```

**Comment by goodmind** [comment]
Duplicate of https://github.com/facebook/flow/issues/4752

**Comment by Bannerets** [comment]
```js
interface JestMockFn<TArguments> {
  (...args: TArguments): void;
}
```

That example seems to be fixed in v0.75.0.
